python/runtime/cudaq/algorithms/py_translate.cpp

```diff
@@ -0,0 +1,128 @@
+#include <pybind11/stl.h>
+
+namespace cudaq {
+
+void pyAltLaunchKernel(const std::string &, MlirModule, OpaqueArguments &,
+                       const std::vector<std::string> &);
+
+std::string getQIRLL(const std::string &name, MlirModule module,
+                     cudaq::OpaqueArguments &runtimeArgs, std::string &profile);
+
+std::string getASM(const std::string &name, MlirModule module,
+                   cudaq::OpaqueArguments &runtimeArgs);
+
+/// @brief Run `cudaq::translate` on the provided kernel.
+std::string pyTranslate(py::object &kernel, py::args args, std::string format) {
+
+  if (py::hasattr(kernel, "compile"))
+    kernel.attr("compile")();
+
+  auto name = kernel.attr("name").cast<std::string>();
+  auto module = kernel.attr("module").cast<MlirModule>();
+
+  auto result =
+      llvm::StringSwitch<std::function<std::string()>>(format)
+          .Case("qir",
+                [&]() {
+                  cudaq::OpaqueArguments args;
+                  std::string profile = "";
+                  return getQIRLL(name, module, args, profile);
+                })
+          .Cases("qir-adaptive", "qir-base",
+                 [&]() {
+                   cudaq::OpaqueArguments args;
+                   return getQIRLL(name, module, args, format);
+                 })
+          .Case("openqasm",
+                [&]() {
+                  if (py::hasattr(kernel, "arguments") &&
+                      py::len(kernel.attr("arguments")) > 0) {
+                    throw std::runtime_error("Cannot translate function with "
+                                             "arguments to OpenQasm 2.0.");
+                  }
+                  cudaq::OpaqueArguments args;
+                  return getASM(name, module, args);
+                })
+          .Default([&]() {
+            throw std::runtime_error("Invalid format to translate to: " +
+                                     format);
+            return "Failed to translate to " + format;
+          })();
+
+  return result;
+}
+
+/// @brief Bind the draw cudaq function
+void bindPyTranslate(py::module &mod) {
+  mod.def(
+      "translate", &pyTranslate, py::arg("kernel"), py::kw_only(),
+      py::arg("format") = "qir",
+      R"#(Return a UTF-8 encoded string representing drawing of the execution
+path, i.e., the trace, of the provided `kernel`.
+
+Args:
+  format (str): format to translate to. Available formats: `qir`, `qir-base`,
+    `qir-adaptive`, `openqasm`.
+  kernel (:class:`Kernel`): The :class:`Kernel` to translate.
+  *arguments (Optional[Any]): The concrete values to evaluate the kernel
+    function at. Leave empty if the kernel doesn't accept any arguments.
+
+Returns:
+  The UTF-8 encoded string of the circuit, without measurement operations.
+
+.. code-block:: python
+
+  # Example
+  import cudaq
+  @cudaq.kernel
+  def bell_pair():
+      q = cudaq.qvector(2)
+      h(q[0])
+      cx(q[0], q[1])
+      mz(q)
+  print(cudaq.translate(bell_pair, format:"qir"))
```

Comment:
Does Python accept the `:` notation?
```suggestion
  print(cudaq.translate(bell_pair, format="qir"))
```